modules/parser/src/test/scala/com/gsk/kg/sparqlparser/QuerySamplesTestSpec.scala

```diff
@@ -0,0 +1,185 @@
+        assert(vs.nonEmpty && vs.head == VARIABLE("?parent"))
+      case _ =>
+        fail
+    }
+  }
+
+  "Get parent class with filter" should "parse" in {
+    val query = QuerySamples.q6
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, Filter(funcs,r)) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?parent"))
+      case _ =>
+        fail
+    }
+  }
+
+  "Test multiple hops" should "parse" in {
+    val query = QuerySamples.q7
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, BGP(triples)) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?species"))
+        assert(triples.size==7)
+      case _ =>
+        fail
+    }
+  }
+
+  "Test multiple hops and prefixes" should "parse" in {
+    val query = QuerySamples.q8
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, BGP(triples)) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?species"))
+        assert(triples.size==7)
+      case _ =>
+        fail
+    }
+  }
+
+  "Test find label" should "parse" in {
+    val query = QuerySamples.q9
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, BGP(triples)) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?label"))
+      case _ =>
+        fail
+    }
+  }
+
+  "Test find parent class" should "parse" in {
+    val query = QuerySamples.q10
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, BGP(triples)) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?parent"))
+        assert(triples.size == 1)
+      case _ =>
+        fail
+    }
+  }
+
+
+  "Tests hops and distinct" should "parse" in {
+    val query = QuerySamples.q11
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Distinct(Select(vs, BGP(triples))) =>
+        assert(vs.nonEmpty && vs.head == VARIABLE("?parent_name"))
+        assert(triples.size == 2)
+      case _ =>
+        fail
+    }
+  }
+
+  "Tests filter and bind" should "parse" in {
+    val query = QuerySamples.q12
+    val expr = QueryConstruct.parseADT(query)
+    expr match {
+      case Select(vs, Filter(funcs,Extend(to,from,BGP(_)))) =>
+        assert(vs.nonEmpty && vs.size == 3)
+        assert(funcs.size == 1)
+        assert(to == VARIABLE("?o"))
+      case _ =>
+        fail
+    }
+  }
+
+  // TODO: Comment out for now, need to work on parser since when BIND in different position Jena generates different representation
+//  "Test BIND in another position in the query" should "parse to same as q12" in {
+//    val query = QuerySamples.q13
+//    val expr = QueryConstruct.parseADT(query)
+//    println(expr)
+//    expr match {
+//      case Select(vs, Filter(funcs,Extend(to,from,BGP(_)))) =>
+//        assert(vs.nonEmpty && vs.size == 3)
+//        assert(funcs.size == 1)
+//        assert(to == VARIABLE("?o"))
+//      case _ =>
+//        fail
+//    }
+//  }
```

Comment:
Maybe we can leave the test as ignored then?

```suggestion
  "Test BIND in another position in the query" should "parse to same as q12" ignore {
    val query = QuerySamples.q13
    val expr = QueryConstruct.parseADT(query)
    println(expr)
    expr match {
      case Select(vs, Filter(funcs,Extend(to,from,BGP(_)))) =>
        assert(vs.nonEmpty && vs.size == 3)
        assert(funcs.size == 1)
        assert(to == VARIABLE("?o"))
      case _ =>
        fail
    }
  }
```